apps/autoscheduler/PerfectHashMap.h

```diff
@@ -0,0 +1,369 @@
+#ifndef PERFECT_HASH_MAP_H
+#define PERFECT_HASH_MAP_H
+
+#include <vector>
+#include <algorithm>
+
+// A specialized hash map used in this file. It can only grow, and it
```

Comment:
TODO: add a test for this class.

Reply:
Adding a test found a bug, so +1 to that.